Add SecTrustSettings bindings

ctz opened the pull request:
This allows a consuming crate to enumerate macos trusted certificates via `SecTrustSettingsCopyCertificates`, and then learn whether they're suitable for use in TLS via `SecTrustSettingsCopyTrustSettings`.

I've done some manual testing of this by twiddling keychain trust settings and observing the results.

Sample calling code is here if you're interested: https://github.com/ctz/rustls-native-certs/blob/master/src/macos.rs

kornelski commented on security-framework/src/trust_settings.rs:
```diff
@@ -0,0 +1,253 @@
+
+    #[test]
+    fn test_list_for_system() {
+        list_for_domain(Domain::System);
```
These aren't real tests. Could you check for some known cert? (some longer-lived one that macOS ships with. It's ok if the test breaks in a few years).

ctz replied:
Good idea. I've added one of these for the Let's Encrypt/ISRG root.

kornelski commented on security-framework/src/trust_settings.rs:
```diff
@@ -0,0 +1,253 @@
+                    Ok(CFArray::<SecCertificate>::wrap_under_create_rule(array_ptr))
+                },
+                err => Err(Error::from_code(err)),
+            }?
```
nit: you can add `?` to `Err` arm directly, so the others won't need to be Ok-wrapped.

kornelski commented on security-framework/src/trust_settings.rs:
```diff
@@ -0,0 +1,253 @@
+            let cert_ptr = cert.as_CFTypeRef() as *mut _;
+            match SecTrustSettingsCopyTrustSettings(cert_ptr, self.domain.into(), &mut array_ptr) {
+                errSecNoTrustSettings => {
+                    Ok(CFArray::from_CFTypes(&[]))
```
The docs say " An empty Trust Settings array is definitely not the same as *no* Trust Settings, which means "this cert must be verified to a known trusted cert". , but an empty array here will fall back to trust!

ctz replied:
Yeah, it seems reasonable to return None here if there were no errors but, equally, no specific settings. I'll do that.

ctz commented:
Thanks for the time you spent reviewing this. I think I've addressed all your comments in some remedial commits; let me know if you prefer these to be squashed.

kornelski commented:
Great. Thank you!
